core/controllers/reader_test.py

```diff
@@ -115,41 +116,47 @@ def setUp(self):
         self.enable_string_classifier = self.swap(
             feconf, 'ENABLE_STRING_CLASSIFIER', True)
 
-        # Reading YAML exploration into a dictionary.
-        yaml_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                 '../tests/data/string_classifier_test.yaml')
-        with open(yaml_path, 'r') as yaml_file:
-            self.yaml_content = yaml_file.read()
-
         self.login(self.VIEWER_EMAIL)
         self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
 
         # Load demo exploration.
-        self.exp_id = '0'
+        self.unused_exp_id = '0'
```

Comment:
You need to load an exploration with this unused id so that the classify handler gets called. It doesn't work without loading one. The id 16 is used because that is the id they used to denote the classifier demo exploration in classifier_services_test.py.